src/MangaBoxParser.ts

```diff
@@ -0,0 +1,208 @@
+    const mangaItems: PartialSourceManga[] = []
+    const collecedIds: string[] = []
+
+    for (const manga of $(source.mangaListSelector).toArray()) {
+        const mangaId = $('a', manga).first().attr('href')
+        const image = $('img', manga).first().attr('src') ?? ''
+        const title = decodeHTML($('a', manga).first().attr('title')?.trim() ?? '')
+        const subtitle = $(source.mangaSubtitleSelector, manga).first().text().trim() ?? ''
+
+        if (!mangaId || !title || collecedIds.includes(mangaId)) continue
+        mangaItems.push(App.createPartialSourceManga({
+            mangaId: mangaId,
+            image: image,
+            title: title,
+            subtitle: subtitle ? subtitle : 'No Chapters'
+        }))
+        collecedIds.push(mangaId)
+    }
+
+    return mangaItems
+}
+
+export const parseMangaDetails = ($: cheerio.Root, mangaId: string, source: MangaBox): SourceManga => {
+    const mangaRootSelector = $(source.mangaRootSelector)
+
+    const image = $(source.mangaThumbnailSelector).attr('src') ?? ''
+
+    const titles = []
+    titles.push(decodeHTML($(source.mangaTitleSelector, mangaRootSelector).text().trim()))
+
+    // Alternative Titles
+    for (const altTitle of $(source.mangaAltTitleSelector, mangaRootSelector)
+        .text()
+        ?.split(/,|;|\//)) {
+        if (altTitle == '') continue
+        titles.push(decodeHTML(altTitle.trim()))
+    }
+
+    const rawStatus = $(source.mangaStatusSelector, mangaRootSelector).text().trim() ?? 'ONGOING'
+    let status = 'ONGOING'
+    switch (rawStatus.toUpperCase()) {
+        case 'ONGOING':
+            status = 'Ongoing'
+            break
+        case 'COMPLETED':
+            status = 'Completed'
+            break
+        default:
+            status = 'Ongoing'
+            break
+    }
+
+    const author = $(source.mangaAuthorSelector, mangaRootSelector)
+        .toArray()
+        .map(x => $(x).text().trim())
+        .join(', ') ?? ''
+
+    const desc = decodeHTML($(source.mangaDescSelector).first().children().remove().end().text().trim())
+
+    let hentai = false
+
+    const tags: Tag[] = []
+    for (const tag of $(source.mangaGenresSelector, mangaRootSelector).toArray()) {
+        const id = $(tag).attr('href')
+        const label = $(tag).text().trim()
+
+        if (!id || !label) continue
+        if (['ADULT', 'SMUT', 'MATURE'].includes(label.toUpperCase())) hentai = true
+
+        tags.push({ id: id, label: label })
+    }
+    const TagSection: TagSection[] = [
+        App.createTagSection({
+            id: '0',
+            label: 'genres',
+            tags: tags.map(t => App.createTag(t)),
+        }),
+    ]
+
+    return App.createSourceManga({
+        id: mangaId,
+        mangaInfo: App.createMangaInfo({
+            image: image,
+            titles: titles,
+            status: status,
+            author: author ? author : 'Unkown',
+            desc: desc,
+            hentai: hentai,
```

Comment:
I don't think the hentai property is required/being used anymore

Reply:
They type definition showed it as optional so wasn't sure. I'll remove it if it's no longer being used.